src/methods.ts

```diff
@@ -254,6 +347,11 @@ export class Method extends AbstractMethodField {
    */
   private code: any;
 
+  private numOpcodeExecs = 0;
+  private jitThreshold = 0;
+  private compiledFunctions: Function[] = [];
```

Comment:
I do have a branch that re-JITs whenever byte code is updated to one of the "fast" opcodes. However, I didn't observe significant performance improvement with it.

My version of re-JIT emits the function source again. Using the previous source is not directly possible here because of lost information about the symbol-stack. We would need to keep that state information but for the first cut I thought it would be too complex.